src/arch/cortex-m0/ctx_switch.S

```diff
@@ -27,29 +27,68 @@ EXC_RETURN_PSP:
   .word 0xFFFFFFFD
 
 .thumb_func
-/* r0 is top of user stack, r1 is heap base */
+/* r0 is top of user stack, r1 Process GOT */
 switch_to_user:
-  /* Load bottom of stack into Process Stack Pointer */
-  msr psp, r0
-
-  /* Cortex-M0 can only push registers R0-R7 directly, so move R8-R11 to R0-R3.
-   * This is equivalent to the 32-bit "push {r4-r11}" instruction. */
-  push {r4-r7}
-  mov r4,  r8
-  mov r5,  r9
-  mov r6, r10
-  mov r7, r11
-  push {r4-r7}
-
-  mov r9, r1
-  svc 0xff
-
-  /* These instructions are equivalent to the 32-bit "pop {r4-r11}" */
-  pop {r4-r7}
-  mov  r8, r4
-  mov  r9, r5
-  mov r10, r6
-  mov r11, r7
-  pop {r4-r7}
-
-  bx lr
+    /* Cortex-M0 can only push registers R0-R7 directly, so move R8-R11 to R0-R3.
+     * This is equivalent to the 32-bit "push {r4-r11}" instruction. */
+    push {r4-r7}
+    mov r4,  r8
+    mov r5,  r9
+    mov r6, r10
+    mov r7, r11
+    push {r4-r7}
+
+    /* Load non-hardware-stacked registers from Process stack */
+    subs r0, #32
+    ldr r7, [r0, #0]
+    ldr r6, [r0, #4]
+    ldr r5, [r0, #8]
+    ldr r4, [r0, #12]
```

Comment:
Any reason not to use `ldmia r0!, {r4-r7}` for continuity with the m4 implementation, and the push right above?